aging_due_report/wizard/wizard.py

```diff
@@ -443,24 +370,18 @@ def _get_lines_by_partner_without_invoice(self, rp_ids):
 
     @api.multi
     def _get_aml(self, aml_ids, inv_type='out_invoice', currency_id=None):
-        aml_obj = self.env['account.move.line']
         res = 0.0
-        sign = 1 if inv_type == 'out_invoice' else -1
         if not aml_ids:
             return res
+        sign = 1 if inv_type == 'out_invoice' else -1
         if currency_id:
-            aml_gen = (
+            res = sum(
                 aml_brw.amount_currency * sign
-                for aml_brw in aml_obj.browse(aml_ids))
-            for iii in aml_gen:
-                res += iii
+                for aml_brw in aml_ids)
         else:
-            aml_gen = (
-                aml_brw.debit and (aml_brw.debit * sign) or
-                aml_brw.credit * (-1 * sign)
-                for aml_brw in aml_obj.browse(aml_ids))
-            for iii in aml_gen:
-                res += iii
+            res = sum(
+                (aml_brw.debit * sign) or aml_brw.credit * (-1 * sign)
```

Comment:
What about use read_group here?

```python
        if currency_id:
            amount = aml_obj.read_group(
                [('id', 'in', aml_ids), ('amount_currency', '!=', 0)],
                ['amount_currency'], [])
            amount = amount[0] if amount else {}
            res = amount.get('amount_currency', 0) * sign
    else:
            amount = aml_obj.read_group(
                [('id', 'in', aml_ids)], ['debit', 'credit'], [])
            amount = amount[0] if amount else {}
            res = (amount.get('debit', 0) * sign +
                   amount.get('credit', 0) * sign * -1)
```

Note:
You can use directly a sum `aml.debit*sign + aml.credit*sign*-1 ` because production data don't have values in both fields at same time.

You can verify it using:
`SELECT debit, credit FROM account_move_line WHERE debit!=0 and credit!=0 LIMIT 1`
(same case for null values)